src/please.go

```diff
@@ -733,7 +736,8 @@ func Please(targets []core.BuildLabel, config *core.Configuration, prettyOutput,
 	state.SetIncludeAndExclude(opts.BuildFlags.Include, opts.BuildFlags.Exclude)
 	parse.InitParser(state)
 	build.Init(state)
-	if config.Events.Port != 0 && shouldBuild {
+
+	if config.Events.Port != 0 && state.NeedBuild {
```

Comment:
guess this doesn't really matter, but these are equivalent right? you assign `state.NeedBuild = shouldBuild` back on line 726.